### PR TITLE
Status indicator devel

### PR DESCRIPTION
**Pull request by jTucker583**
Adds a status indicator widget to DefaulLayoutActivity. This pulls data from PachKeyManager via a coroutine, and publishes it to a widget called PachWidget. That widget displays that data.

Currently, that widget is only set up to take two primitive datatypes, a string and a bool. Those are used for the SAFARI related message that gets displayed, and the controller's connection status to SAFARI, respectively.

### CODE REVIEW

File: SampleCode-V5/android-sdk-v5-uxsdk/src/main/java/dji/v5/ux/pachWidget/PachWidget.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import androidx.lifecycle.LifecycleOwner;
+
+import dji.v5.ux.R;
+import dji.v5.ux.core.base.DJISDKModel;
+import dji.v5.ux.core.base.widget.ConstraintLayoutWidget;
+import dji.v5.ux.core.communication.ObservableInMemoryKeyedStore;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.rxjava3.core.Flowable;
+
+public class PachWidget extends ConstraintLayoutWidget<Object> {
```
Comment (hmray2025) [inline SampleCode-V5/android-sdk-v5-uxsdk/src/main/java/dji/v5/ux/pachWidget/PachWidget.java:21]: Please add comments this class describing it's functionality.

File: SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt [hunk 2]
```diff
@@ -68,7 +76,20 @@ class PachKeyManager() {
...
     }
     // Initialize necessary classes
+    private var pachModel: PachWidgetModel = PachWidgetModel.getInstance()
+    var listener: WaypointListener? = null
     val telemService = TuskServiceWebsocket()
+    private var safetyFailures: Array<Int> = arrayOf(0,0,0,0,0)
+    private var action: String = ""
+    private var autonomous: Boolean = false
+    private var endOfFlight: Boolean = false
+    private val safetyWarnings = mapOf(
```
Comment (hmray2025) [inline SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt:91]: Do you think these strings would be better placed in the "resources" folder? it seems like that was the proper approach towards development.

File: SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt [hunk 3]
```diff
@@ -128,7 +149,50 @@ class PachKeyManager() {
 //        streamer.startStream()
     }
 
+    fun updateStatusWidget() {
+        // Function checks if there are any waypoints in the waypoint list
+        // If there are, it will follow the waypoints
```
Comment (hmray2025) [inline SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt]: Update docstring with more up-to-date information.

File: SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt [hunk 4]
```diff
@@ -836,6 +905,7 @@ class PachKeyManager() {
                 Log.v("PachKeyManagerHIPPO", "Unknown Decision Check Failed")
                 break
             }
+            this@PachKeyManager.action = ""
```
Comment (hmray2025) [inline SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt]: Does this simply reset the action string?

File: SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt [hunk 5]
```diff
@@ -1025,6 +1097,23 @@ class PachKeyManager() {
             return 90-res
         }
     }
+
+    interface WaypointListener {
```
Comment (hmray2025) [inline SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt]: Add documentation.

File: SampleCode-V5/android-sdk-v5-uxsdk/src/main/java/dji/v5/ux/pachWidget/PachWidgetModel.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import androidx.lifecycle.ViewModel;
+
+public class PachWidgetModel extends ViewModel{
+    private static PachWidgetModel instance = null;
+    private IPachWidgetModel pachKeyManager;
+    private MutableLiveData<String> _msgdata;
+    protected LiveData<String> msgdata;
+    private MutableLiveData<Boolean> _connectiondata;
+    protected LiveData<Boolean> connectiondata;
+    private PachWidgetModel() {
```
Comment (hmray2025) [inline SampleCode-V5/android-sdk-v5-uxsdk/src/main/java/dji/v5/ux/pachWidget/PachWidgetModel.java]: If this was built from a previous widget model, please reference that here.

**Review by hmray2025 — CHANGES_REQUESTED** [review]
It generally looks. Please add some more documentation on some of the changes. Otherwise, we should be good to go.

File: SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt (hunk 5 above)
Comment (jTucker583) [reply SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt]: This is something that was left in for the waypoint stuff. I may go with a different implementation, so for now it has just been commented out.

File: SampleCode-V5/android-sdk-v5-uxsdk/src/main/java/dji/v5/ux/pachWidget/PachWidgetModel.java (hunk 6 above)
Comment (jTucker583) [reply SampleCode-V5/android-sdk-v5-uxsdk/src/main/java/dji/v5/ux/pachWidget/PachWidgetModel.java]: Descriptions of why these variables exist have been added.

File: SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt (hunk 4 above)
Comment (jTucker583) [reply SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt]: Yes, when no specific action is being taken, the string is reset to empty.

File: SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt (hunk 2 above)
Comment (jTucker583) [reply SampleCode-V5/android-sdk-v5-sample/app-aircraft/src/main/java/dji/sampleV5/aircraft/control/PachKeyManager.kt:91]: Earmarked for changes after the demo day

**Review by hmray2025 — APPROVED** [review]

**Pull request merged**